libecho/include/EchoProtocol.h

```diff
@@ -42,42 +62,72 @@ class EchoProtocol {
 
     /**
      * @brief Closes a connection.
+     * Connection that was once opened should be closed (even when one of read/write
+     * functions throws an exception).
      */
     void close();
 
     /**
-     * Reads at most count bytes from audio input.
-     * Can receive 0 bytes, but if it doesn't receive any signal for timeout seconds,
-     * it throws an exception.
+     * @brief           Reads at most @p count bytes from audio input and fills buffer with them.
+     * @param buf       pointer to buffer that should be filled
+     * @param count     number of bytes to read
+     * @param timeout   number of seconds; if no signal is detected by then, function assumes
+     *                  that connection is broken; some signals should be continuously sent
+     *                  by protocol to keep connection alive even when no data is being sent;
+     *                  timeout < 0 means that function can wait infinitely
+     *
+     * @return          Returns number of read bytes (can be any from 0 to count),
+     *                  or -1 in case of failure or end of stream.
+     *                  After returning -1, program should call EchoProtocol::close().
      */
     ssize_t read(void *buf, size_t count, int timeout);
 
     /**
-     * Writes count bytes to audio output.
-     * Can throw an exception when sent data isn't being confirmed.
+     * @brief         Writes @p count bytes to audio output.
+     *                Can throw an exception (ConnectionBroken)
+     *                when sent data isn't being confirmed
+     *                (this means some problems in lower layer).
+     *
+     * @param buf     pointer to buffer with data
+     * @param count   number of bytes to send
+     *
+     * @return        Returns number of sent bytes (should be equal to count).
```

Comment:
"should be" - kiedy może się zdarzyć że nie jest i co to oznacza dla użytkownika?